### PR TITLE
Switch to pino for logging with pretty printing

### PR DESCRIPTION
**Pull request by j16r**
Adds friendlier logging in development mode:

Screen Shot 2020-03-06 at 4 23 43 PM

### CODE REVIEW

File: explorer/package.json [hunk 1]
```diff
@@ -32,23 +32,25 @@
   },
   "main": "src/index.ts",
   "dependencies": {
+    "@types/express-pino-logger": "^4.0.2",
```
Comment (HenryNguyen5) [inline explorer/package.json]: Can this be moved to "devDependencies"?

File: explorer/package.json [hunk 2]
```diff
@@ -32,23 +32,25 @@
...
+    "@types/pino": "^5.15.5",
```
Comment (HenryNguyen5) [inline explorer/package.json]: Can this be moved to "devDependencies"?

File: explorer/package.json (hunk 1 above)
Comment (HenryNguyen5) [reply explorer/package.json]: You can do so via
```sh
yarn remove @types/express-pino-logger
yarn add -D @types/express-pino-logger
```

File: explorer/src/database.ts [hunk 3]
```diff
@@ -26,7 +27,7 @@ const loadOptions = (env?: string) => {
 // Loads the following ENV vars, giving them precedence.
 // i.e. TYPEORM_PORT will replace "port" in ormconfig.json.
 const mergeOptions = (): PostgresConnectionOptions => {
-  const envOptions: { [key: string]: string } = {}
+  const envOptions: { [key: string]: any } = { logger: new TypeOrmLogger() }
```
Comment (HenryNguyen5) [inline explorer/src/database.ts]: This isn't really an `envOption` is it? I would probably just specify the logger on line 37 directly inline, then there's no need to mess with the typing of `envOptions`

File: explorer/src/logging.ts [hunk 4]
```diff
@@ -1,46 +1,65 @@
...
 
-const LOGGER_WHITELIST = [
-  'url',
-  'method',
-  'httpVersion',
-  'originalUrl',
-  'query',
-]
-requestWhitelist.splice(0, requestWhitelist.length, ...LOGGER_WHITELIST)
+const options: any = {
```
Comment (HenryNguyen5) [inline explorer/src/logging.ts]:
```suggestion
const options: Parameters<typeof pino>[0] = {
```

File: explorer/src/logging.ts [hunk 5]
```diff
@@ -1,46 +1,65 @@
...
+  name: 'Explorer',
+  level: 'warn',
+  redact: {
+    paths: ['req.headers', 'res.headers'],
+  },
+}
+if (process.env.EXPLORER_DEV) {
+  options['prettyPrint'] = { colorize: true }
```
Comment (HenryNguyen5) [inline explorer/src/logging.ts]: Typically we use dot notation for object access, unless its for keys that have invalid syntax like hyphens.

```suggestion
  options.prettyPrint = { colorize: true }
```

File: explorer/src/logging.ts [hunk 6]
```diff
@@ -1,46 +1,65 @@
...
+    logger.trace({msg: message})
+  }
+
+  public logMigration(message: string, queryRunner?: QueryRunner): any {
+    logger.info({msg: message})
+  }
+
+  public log(
+    level: 'log' | 'info' | 'warn',
+    message: any,
```
Comment (HenryNguyen5) [inline explorer/src/logging.ts]: I would restrict this to `string | number | null | undefined` or something like that. Passing an object to a string templates results in it being cased to `[object Object]'

File: explorer/package.json (hunk 1 above)
Comment (j16r) [reply explorer/package.json]: No, then we would have no request and response logging :(

File: explorer/package.json (hunk 2 above)
Comment (j16r) [reply explorer/package.json]: No, then we'd have no logging at all.

File: explorer/package.json (hunk 1 above)
Comment (HenryNguyen5) [reply explorer/package.json]: This is just moving the type definition to dev dependencies. It's not used during runtime.

File: explorer/package.json (hunk 2 above)
Comment (HenryNguyen5) [reply explorer/package.json]: This is just moving the type definition to dev dependencies. It's not used during runtime.

File: explorer/src/logging.ts (hunk 6 above)
Comment (j16r) [reply explorer/src/logging.ts]: The original interface is `any` so changing this would have no effect (not sure if it'd be accepted either).

**Pull request merged**